services/src/main/java/io/druid/cli/PullDependencies.java

```diff
@@ -19,33 +19,76 @@
 
 import com.google.common.base.Throwables;
 import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
 import com.google.inject.Inject;
+import com.metamx.common.StringUtils;
+import com.metamx.common.logger.Logger;
 import io.airlift.command.Command;
 import io.airlift.command.Option;
 import io.druid.guice.ExtensionsConfig;
 import io.druid.indexing.common.config.TaskConfig;
-import io.druid.initialization.Initialization;
+import io.tesla.aether.Repository;
 import io.tesla.aether.internal.DefaultTeslaAether;
+import org.apache.commons.io.FileUtils;
+import org.eclipse.aether.artifact.Artifact;
+import org.eclipse.aether.artifact.DefaultArtifact;
+import org.eclipse.aether.collection.CollectRequest;
+import org.eclipse.aether.graph.Dependency;
+import org.eclipse.aether.graph.DependencyFilter;
+import org.eclipse.aether.graph.DependencyNode;
+import org.eclipse.aether.resolution.DependencyRequest;
+import org.eclipse.aether.util.artifact.JavaScopes;
+import org.eclipse.aether.util.filter.DependencyFilterUtils;
 
+import java.io.File;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.io.PrintStream;
+import java.io.UnsupportedEncodingException;
+import java.net.URI;
+import java.net.URISyntaxException;
 import java.util.List;
+import java.util.Set;
 
 
 @Command(
     name = "pull-deps",
-    description = "Pull down dependencies to the local repository specified by druid.extensions.localRepository"
+    description = "Pull down dependencies to the local repository specified by druid.extensions.localRepository, extensions directory specified by druid.extensions.extensionsDir and hadoop depenencies directory specified by druid.extensions.hadoopDependenciesDir"
```

Comment:
@xvrl I think local repository still serves as a cache for downloading maven coordinates, so that if you call ```tool-deps``` multiple times for the same maven coordinate, only the first time will actually need to download files from internet, which takes a bit time to finish.